### PR TITLE
gvfs: update to 1.54.0, recursive cleanup

### PR DESCRIPTION
**Pull request by cinerea0**
#### Testing the changes
- I tested the changes in this PR: **YES**

---

I tested that gvfs works properly. However, I don't use OneDrive, so I can't verify that part works correctly.

### CODE REVIEW

**Comment by chrysos349** [comment]
`libgdata-0.18.1` fails to build with the latest `uhttmock-0.10.0`, because it needs the old uhttmock and libsoup2.

```diff
diff --git a/srcpkgs/libgdata/template b/srcpkgs/libgdata/template
index 835f82f7f..9d5ee2c28 100644
--- a/srcpkgs/libgdata/template
+++ b/srcpkgs/libgdata/template
@@ -1,13 +1,13 @@
 # Template file for 'libgdata'
 pkgname=libgdata
 version=0.18.1
-revision=1
+revision=2
 build_helper="gir"
 build_style=meson
 configure_args="-Dintrospection=$(vopt_if gir true false)
- -Dvapi=$(vopt_if vala true false) -Dgtk_doc=false"
+ -Dvapi=$(vopt_if vala true false) -Dgtk_doc=false -Dalways_build_tests=false"
 hostmakedepends="pkg-config flex intltool glib-devel $(vopt_if vala vala)"
-makedepends="json-glib-devel gtk+3-devel uhttpmock-devel liboauth-devel
+makedepends="json-glib-devel gtk+3-devel liboauth-devel
  gcr-devel gnome-online-accounts-devel"
 short_desc="Library for accessing GData webservices"
 maintainer="Enno Boland <gottox@voidlinux.org>"
```

**Comment by oreo639** [comment]
Should the onedrive backend go with gvfs-goa? (the google backend is shipped in the gvfs-goa package)

**Comment by cinerea0** [comment]
I'm not as familiar with the GnomeOnlineAccounts ecosystem, so I'm not sure. Their [providers page](https://wiki.gnome.org/Projects/GnomeOnlineAccounts/Providers) doesn't mention files for Microsoft.

**Comment by oreo639** [comment]
It was added in GNOME 46.

**Comment by cinerea0** [comment]
Just made a new push with OneDrive added to the GOA subpackage.

File: srcpkgs/msgraph/template [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+configure_args="$(vopt_bool gtk_doc gtk_doc)"
+hostmakedepends="pkg-config $(vopt_if gtk_doc gi-docgen)"
+makedepends="glib-devel gnome-online-accounts-devel json-glib-devel rest-devel
+ uhttpmock-devel"
+short_desc="Library to access MS Graph API for Office 365"
+maintainer="cinerea0 <cinerea0@protonmail.com>"
+license="LGPL-3.0-or-later"
+homepage="https://gitlab.gnome.org/GNOME/msgraph"
+changelog="https://gitlab.gnome.org/GNOME/msgraph/-/raw/main/NEWS"
+distfiles="https://gitlab.gnome.org/GNOME/msgraph/-/archive/${version}/msgraph-${version}.tar.gz"
```
Comment (fanyx) [inline srcpkgs/msgraph/template:16]:
```suggestion
distfiles="${GNOME_SITE}/msgraph/${version%.*}/msgraph-${version}.tar.xz"
```

**Pull request merged**